packages/c/sshnpd/src/permitopen.c

```diff
@@ -0,0 +1,101 @@
+#include "sshnpd/permitopen.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/errno.h>
+
+int parse_permitopen(char *input, char ***permitopen_hosts, uint16_t **permitopen_ports, size_t *permitopen_len,
+                     bool is_logger_available) {
+  const char *TAG = "parse_permitopen";
+  int sep_count = 0;
+  int permitopen_end = strlen(input);
+
+  for (int i = 0; i < permitopen_end; i++) {
+    if (input[i] == ':') {
+      sep_count++;
+      input[i] = '\0';
+    }
+
+    if (input[i] == ',') {
+      input[i] = '\0';
+    }
+  }
+
+  // malloc pointers to each string, but don't malloc any more memory for individual char storage
+  *permitopen_hosts = malloc((sep_count) * sizeof(char *));
+  *permitopen_ports = malloc((sep_count) * sizeof(uint16_t));
```

Comment:
we don't need individual char storage because we are pointing these pointers to the memory passed into the argv of the program.